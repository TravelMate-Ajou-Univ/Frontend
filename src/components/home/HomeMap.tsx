"use client";

import { ArticleCountType, SeasonType } from "@/model/article";
import { useEffect, useState } from "react";
import SeasonNav from "../ui/SeasonNav";
import Image from "next/image";
import map from "/public/image/map.png";
import { articleCount } from "@/service/axios/article";
import _ from "lodash";
import Link from "next/link";

const ARTICLE_NUM_CLASS =
  "rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer";

export default function HomeMap() {
  const [season, setSeason] = useState<SeasonType>("SPRING");
  const [articleCounts, setArticleCounts] = useState<ArticleCountType[]>([]);

  useEffect(() => {
    const getArticleCounts = async () => {
      const data = await articleCount(season);
      if (!data)
        return alert("데이터를 불러오는데 실패했습니다. 다시 시도해주세요.");
      setArticleCounts(data);
    };

    getArticleCounts();
  }, [season]);

  const selectSeason = (season: SeasonType) => {
    setSeason(season);
  };

  const getArticleCount = (location: string) => {
    return _.find(articleCounts, item => item.location === location)?.count;
  };

  return (
    <section className="flex flex-col items-center">
      <SeasonNav season={season} onClick={selectSeason} />
      <div className="relative mt-2 w-2/3">
        <Image
          className="w-full"
          src={map}
          alt="지도"
          width={10000}
          height={10000}
        />
        <Link
          href={`/article/list/?location=서울&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[20%] left-[25%]"}>
            {getArticleCount("서울")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=경기/인천&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[26%] left-[32%]"}>
            {getArticleCount("경기/인천")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=강원&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[16%] left-[55%]"}>
            {getArticleCount("강원")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=충청/대전&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[40%] left-[30%]"}>
            {getArticleCount("충청/대전")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=경북/대구&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[45%] left-[65%]"}>
            {getArticleCount("경북/대구")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=전라/광주&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[64%] left-[23%]"}>
            {getArticleCount("전라/광주")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=경남/울산/부산&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[65%] left-[53%]"}>
            {getArticleCount("경남/울산/부산")}
          </span>
        </Link>
        <Link
          href={`/article/list/?location=제주&seasons=${season.toLowerCase()}`}
        >
          <span className={ARTICLE_NUM_CLASS + " top-[92.5%] left-[21%]"}>
            {getArticleCount("제주")}
          </span>
        </Link>
      </div>
    </section>
  );
}
