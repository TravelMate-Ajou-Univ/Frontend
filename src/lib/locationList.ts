export const locationList = [
  "서울",
  "경기/인천",
  "강원",
  "충청/대전",
  "전라/광주",
  "경북/대구",
  "경남/울산/부산",
  "제주"
];
