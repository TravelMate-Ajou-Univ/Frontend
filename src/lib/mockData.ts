export const articleMock = {
  id: 1,
  title: "",
  thumbnail: null,
  location: "경기/인천",
  authorId: 1,
  springVersionID: 1,
  summerVersionID: null,
  fallVersionID: null,
  winterVersionID: null,
  articleTagMap: [
    {
      id: 228,
      articleId: 20,
      tagId: 8,
      tag: {
        id: 8,
        name: "카페",
        createdAt: "2023-11-05T11:13:46.000Z",
        updatedAt: "2023-11-05T11:13:46.000Z"
      }
    },
    {
      id: 229,
      articleId: 20,
      tagId: 104,
      tag: {
        id: 104,
        name: "행궁",
        createdAt: "2023-12-08T07:49:33.000Z",
        updatedAt: "2023-12-08T07:49:33.000Z"
      }
    },
    {
      id: 230,
      articleId: 20,
      tagId: 60,
      tag: {
        id: 60,
        name: "행궁 카페",
        createdAt: "2023-11-23T05:46:55.000Z",
        updatedAt: "2023-11-23T05:46:55.000Z"
      }
    }
  ],
  spring: null,
  summer: {
    id: 1,
    articleId: 1,
    userId: 1,
    period: "SUMMER",
    content: "<p>hi</p>",
    createdAt: "2023-12-02T09:31:38.000Z",
    updatedAt: "2023-12-02T09:31:38.000Z"
  },
  fall: null,
  winter: null,
  articleBookmarkMap: [
    {
      id: 90,
      period: "WINTER",
      articleId: 20,
      bookmarkId: 227,
      createdAt: "2023-12-20T14:07:44.000Z",
      updatedAt: "2023-12-20T14:07:44.000Z",
      bookmark: {
        id: 227,
        content: "4층까지 있고 넓고 쾌적",
        locationId: 90,
        userId: 3,
        createdAt: "2023-12-20T14:02:38.000Z",
        deletedAt: null,
        location: {
          id: 90,
          latitude: "37.2848574",
          longitude: "127.0144045",
          placeId: "ChIJ9_ndU51DezURyWyJWMNsnKs"
        }
      }
    }
  ],
  isFavorite: false
};

export const searchedKeywordMock = [
  {
    id: 2,
    name: "제주 맛집",
    createdAt: "2023-11-05T11:00:18.000Z",
    updatedAt: "2023-11-05T11:00:18.000Z"
  },
  {
    id: 34,
    name: "맛집",
    createdAt: "2023-11-08T12:43:01.000Z",
    updatedAt: "2023-11-08T12:43:01.000Z"
  }
];
