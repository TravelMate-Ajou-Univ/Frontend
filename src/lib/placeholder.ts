const KEYWORD_INPUT = "#맛집, #카페, #숙소 등의 키워드를 입력해주세요 :)";

export const placeholder = {
  keywordInput: KEYWORD_INPUT
};
