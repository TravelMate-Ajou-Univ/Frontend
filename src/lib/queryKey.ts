const ARTICLE = "article";
const ARTICLE_EDIT = "article-edit";
const ARTICLE_COUNT = "article-count";

const KEYWORD = "keyword";

export const queryKey = {
  article: ARTICLE,
  articleEdit: ARTICLE_EDIT,
  articleCount: ARTICLE_COUNT,
  keyword: KEYWORD
};
