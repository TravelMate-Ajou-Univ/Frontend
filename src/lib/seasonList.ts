export const seasonList = ["봄", "여름", "가을", "겨울"];

export const seasonMapper = {
  봄: "SPRING",
  여름: "SUMMER",
  가을: "FALL",
  겨울: "WINTER"
};
