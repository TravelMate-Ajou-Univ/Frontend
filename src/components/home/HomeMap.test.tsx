import { composeStories } from "@storybook/react";
import * as stories from "./HomeMap.stories";
import { axiosMock } from "@/test/axios-mock";
import { articleCountMock } from "@/lib/mockData";
import { waitFor } from "@testing-library/react";
import customRender from "@/test/customRender";

const SEASON = "SPRING";

const { Default: HomeMap } = composeStories(stories);

beforeEach(() => {
  axiosMock
    .onGet("articles/count", { params: { period: SEASON } })
    .reply(200, articleCountMock);
});

afterEach(() => {
  axiosMock.reset();
});

describe("<HomeMap />", () => {
  it("지역별 게시글 수를 보여준다.", async () => {
    const component = customRender(<HomeMap />);

    await waitFor(() => {
      expect(component.getAllByText("3")).toHaveLength(2);
      expect(component.getAllByText("0")).toHaveLength(5);
      expect(component.getAllByText("1")).toHaveLength(1);
    });

    expect(component).toMatchSnapshot();
  });
});
