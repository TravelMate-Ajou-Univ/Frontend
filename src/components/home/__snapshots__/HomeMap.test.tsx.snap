// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeMap /> 지역별 게시글 수를 보여준다. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex flex-col items-center"
      >
        <ul
          class="flex justify-around items-center lg:gap-4 md:gap-3 gap-2"
        >
          <li
            class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm border-spring bg-spring"
          >
            <img
              alt="봄 아이콘"
              class="md:w-7 md:h-7 w-5 h-5"
              data-nimg="1"
              decoding="async"
              height="25"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="25"
            />
            <span>
              봄
            </span>
          </li>
          <li
            class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
          >
            <img
              alt="여름 아이콘"
              class="md:w-7 md:h-7 w-5 h-5"
              data-nimg="1"
              decoding="async"
              height="25"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="25"
            />
            <span>
              여름
            </span>
          </li>
          <li
            class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
          >
            <img
              alt="가을 아이콘"
              class="md:w-7 md:h-7 w-5 h-5"
              data-nimg="1"
              decoding="async"
              height="25"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="25"
            />
            <span>
              가을
            </span>
          </li>
          <li
            class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
          >
            <img
              alt="겨울 아이콘"
              class="md:w-7 md:h-7 w-5 h-5"
              data-nimg="1"
              decoding="async"
              height="25"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="25"
            />
            <span>
              겨울
            </span>
          </li>
        </ul>
        <div
          class="relative mt-2 w-2/3"
        >
          <img
            alt="지도"
            class="w-full"
            data-nimg="1"
            decoding="async"
            height="10000"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=3840&q=75 1x"
            style="color: transparent;"
            width="10000"
          />
          <a
            href="/article/list?location=서울&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[20%] left-[25%]"
            >
              3
            </span>
          </a>
          <a
            href="/article/list?location=경기/인천&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[26%] left-[32%]"
            >
              3
            </span>
          </a>
          <a
            href="/article/list?location=강원&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[16%] left-[55%]"
            >
              0
            </span>
          </a>
          <a
            href="/article/list?location=충청/대전&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[40%] left-[30%]"
            >
              0
            </span>
          </a>
          <a
            href="/article/list?location=경북/대구&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[45%] left-[65%]"
            >
              0
            </span>
          </a>
          <a
            href="/article/list?location=전라/광주&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[64%] left-[23%]"
            >
              0
            </span>
          </a>
          <a
            href="/article/list?location=경남/울산/부산&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[65%] left-[53%]"
            >
              1
            </span>
          </a>
          <a
            href="/article/list?location=제주&seasons=spring"
          >
            <span
              class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[92.5%] left-[21%]"
            >
              0
            </span>
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex flex-col items-center"
    >
      <ul
        class="flex justify-around items-center lg:gap-4 md:gap-3 gap-2"
      >
        <li
          class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm border-spring bg-spring"
        >
          <img
            alt="봄 아이콘"
            class="md:w-7 md:h-7 w-5 h-5"
            data-nimg="1"
            decoding="async"
            height="25"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="25"
          />
          <span>
            봄
          </span>
        </li>
        <li
          class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
        >
          <img
            alt="여름 아이콘"
            class="md:w-7 md:h-7 w-5 h-5"
            data-nimg="1"
            decoding="async"
            height="25"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="25"
          />
          <span>
            여름
          </span>
        </li>
        <li
          class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
        >
          <img
            alt="가을 아이콘"
            class="md:w-7 md:h-7 w-5 h-5"
            data-nimg="1"
            decoding="async"
            height="25"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="25"
          />
          <span>
            가을
          </span>
        </li>
        <li
          class="flex justify-center items-center md:gap-2 gap-1 md:py-2.5 sm:py-2 py-1.5 md:w-40 sm:w-28 w-20 cursor-pointer rounded-full border text-gray-900 md:text-base text-sm bg-white"
        >
          <img
            alt="겨울 아이콘"
            class="md:w-7 md:h-7 w-5 h-5"
            data-nimg="1"
            decoding="async"
            height="25"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="25"
          />
          <span>
            겨울
          </span>
        </li>
      </ul>
      <div
        class="relative mt-2 w-2/3"
      >
        <img
          alt="지도"
          class="w-full"
          data-nimg="1"
          decoding="async"
          height="10000"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=3840&q=75 1x"
          style="color: transparent;"
          width="10000"
        />
        <a
          href="/article/list?location=서울&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[20%] left-[25%]"
          >
            3
          </span>
        </a>
        <a
          href="/article/list?location=경기/인천&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[26%] left-[32%]"
          >
            3
          </span>
        </a>
        <a
          href="/article/list?location=강원&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[16%] left-[55%]"
          >
            0
          </span>
        </a>
        <a
          href="/article/list?location=충청/대전&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[40%] left-[30%]"
          >
            0
          </span>
        </a>
        <a
          href="/article/list?location=경북/대구&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[45%] left-[65%]"
          >
            0
          </span>
        </a>
        <a
          href="/article/list?location=전라/광주&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[64%] left-[23%]"
          >
            0
          </span>
        </a>
        <a
          href="/article/list?location=경남/울산/부산&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[65%] left-[53%]"
          >
            1
          </span>
        </a>
        <a
          href="/article/list?location=제주&seasons=spring"
        >
          <span
            class="rounded-full md:px-5 sm:px-4 px-3 py-1 md:py-0.5 border absolute bg-white hover:bg-gray-100 md:text-base sm:text-sm text-xs cursor-pointer top-[92.5%] left-[21%]"
          >
            0
          </span>
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
